virtool_core/models/group.py

```diff
@@ -22,3 +24,13 @@ class GroupMinimal(BaseModel):
 
 class Group(GroupMinimal):
     permissions: Permissions
+    name: Optional[str]
+
+    @validator("name", "id", always=True)
+    def check_name(cls, name, _id):
+        """
+        Sets `name` to the provided `id` if it is none.
+        """
+        if name is None:
+            name = _id
+        return name
```

Comment:
This is sufficient if we disallow or don't want empty strings for `name`. Empty strings evaluate `False` in Python.
```suggestion
        return name or _id
```